monetio/sat/nesdis_viirs_aod_aws_gridded.py

```diff
@@ -0,0 +1,319 @@
+    Parameters:
+    satellite (str): The satellite name. Can be 'both', 'SNPP', or 'NOAA20'.
+    date_generated (list): A list of observation dates.
+    fs (FileSystem): The file system object.
+
+    Returns:
+    tuple: A tuple containing the list of files and the total size of files.
+    """
+    # Loop through observation dates & check for files
+    nodd_file_list = []
+    nodd_total_size = 0
+    for date in date_generated:
+        file_date = date.strftime("%Y%m%d")
+        year = file_date[:4]
+        if satellite == "both":
+            sat_list = ["SNPP", "NOAA20"]
+            for sat_name in sat_list:
+                prod_path = (
+                    "noaa-jpss/"
+                    + sat_name
+                    + "/VIIRS/"
+                    + sat_name
+                    + "_VIIRS_Aerosol_Optical_Depth_Gridded_Reprocessed/0.25_Degrees_Weekly/"
+                    + year
+                    + "/"
+                )
+                # Get list of all files in given year on NODD
+                all_files = fs.ls(prod_path)
+                # Loop through files, check if file date falls within observation date range
+                for file in all_files:
+                    file_start = file.split("/")[-1].split("_")[7].split(".")[0].split("-")[0]
+                    file_end = file.split("/")[-1].split("_")[7].split(".")[0].split("-")[1]
+                    # If file within observation range, add path to list and add file size to total
+                    if file_date >= file_start and file_date <= file_end:
+                        if file not in nodd_file_list:
+                            nodd_file_list.append(file)
+                            nodd_total_size = nodd_total_size + fs.size(file)
+        else:
+            prod_path = (
+                "noaa-jpss/"
+                + satellite
+                + "/VIIRS/"
+                + satellite
+                + "_VIIRS_Aerosol_Optical_Depth_Gridded_Reprocessed/0.25_Degrees_Weekly/"
+                + year
+                + "/"
+            )
+            # Get list of all files in given year on NODD
+            all_files = fs.ls(prod_path)
+            # Loop through files, check if file date falls within observation date range
+            for file in all_files:
+                file_start = file.split("/")[-1].split("_")[7].split(".")[0].split("-")[0]
+                file_end = file.split("/")[-1].split("_")[7].split(".")[0].split("-")[1]
+                # If file within observation range, add path to list and add file size to total
+                if file_date >= file_start and file_date <= file_end:
+                    if file not in nodd_file_list:
+                        nodd_file_list.append(file)
+                        nodd_total_size = nodd_total_size + fs.size(file)
+
+    return nodd_file_list, nodd_total_size
+
+
+def open_dataset(
+    date, satellite, data_resolution="0.1", averaging_time="daily", download=False, save_path="./"

```

Comment:
```suggestion
    date, satellite, data_resolution="0.100", averaging_time="daily", download=False, save_path="./"

```
one of the default options